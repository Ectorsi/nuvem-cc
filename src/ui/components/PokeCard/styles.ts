import styled from 'styled-components';
import Theme from '../../../styles/theme';

export const Container = styled.button`
    background-color: ${Theme.colors.grayScale.bgCard};
    height: 50px;
    border: none;
    border-bottom: 1px solid #c4c4c4;
    width: 100%;

    &:hover {
        background-color: ${Theme.colors.grayScale.bgCardHover};
    }

    display: flex;
    align-items: center;
    justify-content: flex-start;
    padding: 0 10px;

    color: ${Theme.colors.font.primary};
    font-weight: 500;
    letter-spacing: 1px;
`;
