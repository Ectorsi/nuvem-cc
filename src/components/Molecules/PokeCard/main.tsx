import React from "react";
import * as S from "./styles";

type PokeCardProps = {
    name: string,
    rest?: any,
}

const PokeCard = ({ name, ...rest }: PokeCardProps) => {
    return (
        <S.Container type="button" {...rest}>
            {name}
        </S.Container>
    )
};

export default PokeCard;