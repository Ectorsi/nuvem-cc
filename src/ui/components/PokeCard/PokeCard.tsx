import { memo } from 'react';
import * as S from './styles';

export type PokeCardProps = {
    name: string;
    handleSelectPokemon: (pokemonName: string) => void;
};

const PokeCard = ({ name, handleSelectPokemon }: PokeCardProps) => (
    <S.Container
        type="button"
        onClick={() => {
            handleSelectPokemon(name);
        }}
    >
        <h1>{name}</h1>
    </S.Container>
);

export default memo(PokeCard);
