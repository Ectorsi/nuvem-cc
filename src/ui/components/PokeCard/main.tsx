import React from "react";
import * as S from "./styles";

export type PokeCardProps = {
    name: string;
    handleShowModal: () => void
    handleSelectPokemon: (pokemonName: string) => void
};

const PokeCard = ({ name, handleShowModal, handleSelectPokemon }: PokeCardProps) => (
    <S.Container type="button" onClick={() => {
        handleSelectPokemon(name);
        handleShowModal();
    }}>
        <h1>{name}</h1>
    </S.Container>
);

export default PokeCard;