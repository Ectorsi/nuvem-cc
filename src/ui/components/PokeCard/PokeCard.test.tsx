import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import PokeCard from './PokeCard';

const mockPokeCard = {
    name: "pikachu",
    handleShowModal: jest.fn(),
    handleSelectPokemon: jest.fn(),
};

describe('PokeCard unit test', () => {
    it('should be able to show the pokemon name', async () => {
        const { container } = render(<PokeCard {...mockPokeCard} />);
        const name = screen.getByText(/pikachu/i);
        expect(name).toBeInTheDocument();
        expect(container.firstChild).toMatchSnapshot();
    });

    it('should be able to click and open modal and then, close it', async () => {
        render(<PokeCard {...mockPokeCard} />);
        const cardName = screen.getByText(/pikachu/i);
        fireEvent.click(cardName);
        expect(cardName).toBeInTheDocument();
    });
});