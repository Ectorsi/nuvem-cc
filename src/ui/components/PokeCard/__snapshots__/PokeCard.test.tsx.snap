// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokeCard unit test should be able to show the pokemon name 1`] = `
<button
  class="sc-beqWaB dQYWhm"
  type="button"
>
  <h1>
    pikachu
  </h1>
</button>
`;
